src/components/GenreBar/GenreBar.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import './style.scss';
+import { GenreBarItem } from 'components/GenreBar/GenreBarItem/GenreBarItem';
+import { GENRE_BAR } from 'utils/constant';
+
+const GenreBar = () => {
+  const [activeLineStyle, setActiveLineStyle] = useState(null);
+  const [activeGenreItem, selectActiveGenreItem] = useState(GENRE_BAR[0]);
+
+  const handleClick = (target: HTMLElement, newLabel: string) => {
+    selectActiveGenreItem({ label: newLabel });
+    setActiveLineStyle({
+      width: target.offsetWidth,
+      marginLeft: target.offsetLeft
+    });
+  };
+
+  const genreList = GENRE_BAR.map(({ label }) => (
+    <GenreBarItem
+      label={label}
+      isActive={label === activeGenreItem?.label}
+      key={label}
+      handleClick={target => handleClick(target, label)}
```

Comment:
Please consider using useCallback react hook for callbacks

Reply:
useCallback will added in the next pull request